tests/unit/test_general_functions.py

```diff
@@ -530,31 +533,144 @@ def test_all_digits_in_string_are_superscripted(self):
 
 class TestUpdatingReferenceDensity(unittest.TestCase):
 
-    # TODO: More tests
     def test_update_reference_density(self):
 
-        target = mo.Target(layers=[
+        no_layers = mo.Target(layers=[])
```

Comment:
For improved readablity, consider reorganizing the test like this:

```python
        target_no_layers = mo.Target(layers=[])
        no_layers = ReferenceDensity(target_no_layers.layers)
        no_layers.update_reference_density()
        self.assertEqual(no_layers.reference_density, 0.0)

        target_one_layer_under_limit = mo.Target(layers=[
            Layer("Si", [
                Element.from_string("Si 1.0")
            ], 1.0, 2.3290, start_depth=0.0)
        ])
        one_layer_under_limit = ReferenceDensity(
            target_one_layer_under_limit.layers)
        one_layer_under_limit.update_reference_density()
        self.assertEqual(one_layer_over_limit.reference_density,
                         4.9897601328705675e+22)
```

Also note the renamed variables (target_**name**, **name**).

Reply:
I'm not sure how this improves readability?

Reply:
OK